core/src/extension.rs

```diff
@@ -0,0 +1,70 @@
+use crate::plugin::Plugin;
+use crate::uri::{Uri, UriBound};
+use std::any::Any;
+use std::marker::PhantomData;
+use std::os::raw::c_void;
+
+/// A trait for marking a type as an LV2 Plugin extension.
+///
+/// # Unsafety
+///
+/// The `RAW_DATA` field can be set to any static value, it is up to the implementer of the
+/// extension to set it correctly:
+///
+/// * The reference must point to a structure with the exact same fields as the one defined by the
+///   LV2 Extension specification;
+/// * The struct being pointed to must be `'static`, i.e. available forever;
+/// * The struct being pointed to must be correctly initialized, as defined by the LV2 Extension
+///   specification;
+/// * The struct being pointed to must be `#[repr(C)]`.
+/// * The URI associated to this extension must be the exact same as the one defined in the LV2
+///   Extension specification, and must also be the one tied to the struct being pointed to by
+///   `RAW_DATA`.
+///
+/// Failing to meet any of these requirements will lead to undefined behavior when the host will
+/// try to load the extension for any given plugin.
+///
+pub unsafe trait Extension<P: Plugin>: UriBound {
+    /// The raw data structure defined by the extension, as returned by the plugin's `extension_data()` method.
+    const RAW_DATA: &'static (Any + 'static);
```

Comment:
I took a look at the results from Travis and it looks like using `Any` instead of `dyn Any` is going to be deprecated soon. Maybe you should change it like this:
```suggestion
    const RAW_DATA: &'static (dyn Any + 'static);
```

Reply:
That's odd, i didn't see those warnings locally nor in Travis, but you're right though! I applied your suggestions. :slightly_smiling_face: